### PR TITLE
switch from sapm to otlphttp as default

### PR DESCRIPTION
**Pull request by asreehari-splunk**
**Description:** 
This PR removes sapm from the default configuration, and use otlphttp instead. The sapm exporter has been deprecated in an earlier release.

### CODE REVIEW

File: cmd/otelcol/config/collector/agent_config.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@
 # - SPLUNK_HEC_URL: The Splunk HEC endpoint URL, e.g. https://ingest.us0.signalfx.com/v1/log
 # - SPLUNK_INGEST_URL: The Splunk ingest URL, e.g. https://ingest.us0.signalfx.com
 # - SPLUNK_LISTEN_INTERFACE: The network interface the agent receivers listen on.
-# - SPLUNK_TRACE_URL: The Splunk trace endpoint URL, e.g. https://ingest.us0.signalfx.com/v2/trace
+# - SPLUNK_TRACE_URL: The Splunk trace endpoint URL, e.g. https://ingest.us0.signalfx.com/v2/trace/otlp
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/agent_config.yaml:13]: Why is this listed if it's not being used?

File: .github/workflows/scripts/win-test-services.ps1 [hunk 2]
```diff
@@ -44,7 +44,8 @@ $expected_svc_env_vars = @{
   "SPLUNK_REALM"            = "$realm";
   "SPLUNK_API_URL"          = "$api_url";
   "SPLUNK_INGEST_URL"       = "$ingest_url";
-  "SPLUNK_TRACE_URL"        = "${ingest_url}/v2/trace";
+  # We no longer test for this as we have moved to OTLP and it breaks upgrade tests.
+  #"SPLUNK_TRACE_URL"        = "${ingest_url}/v2/trace/otlp";
```
Comment (dmitryax) [inline .github/workflows/scripts/win-test-services.ps1:48]: let's just remove it

File: cmd/otelcol/config/collector/gateway_config.yaml [hunk 3]
```diff
@@ -125,6 +125,11 @@ exporters:
     token: "${SPLUNK_ACCESS_TOKEN}"
     endpoint: "${SPLUNK_INGEST_URL}/v1/log"
     log_data_enabled: false
+  otlphttp:
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/gateway_config.yaml:128]: let's remove sapm exporter form the config

File: deployments/ansible/molecule/default/Dockerfile.j2 [hunk 4]
```diff
@@ -28,7 +28,7 @@ RUN zypper -n clean && zypper -n refresh
 RUN zypper -n install -l ansible dbus-1 rpm-python sudo systemd-sysvinit
 {% else %}
 FROM opensuse/leap:15
-RUN sed -i 's|download.opensuse.org|provo-mirror.opensuse.org|' /etc/zypp/repos.d/*.repo
+#RUN sed -i 's|download.opensuse.org|provo-mirror.opensuse.org|' /etc/zypp/repos.d/*.repo
```
Comment (dmitryax) [inline deployments/ansible/molecule/default/Dockerfile.j2:31]: Why commented?

File: deployments/ansible/molecule/custom_vars/windows-verify.yml [hunk 5]
```diff
@@ -8,7 +8,7 @@
       SPLUNK_CONFIG: '{{ ansible_env.ProgramData }}\Splunk\OpenTelemetry Collector\custom_config.yml'
       SPLUNK_INGEST_URL: https://fake-splunk-ingest.com
       SPLUNK_API_URL: https://fake-splunk-api.com
-      SPLUNK_TRACE_URL: https://fake-splunk-ingest.com/v2/trace
+      SPLUNK_TRACE_URL: https://fake-splunk-ingest.com/v2/trace/otlp
```
Comment (dmitryax) [inline deployments/ansible/molecule/custom_vars/windows-verify.yml:11]: We don't use `SPLUNK_TRACE_URL`  anymore right?

File: deployments/ansible/molecule/default/verify.yml [hunk 6]
```diff
@@ -45,7 +45,7 @@
 
     - name: Assert SPLUNK_TRACE_URL env var is set
       ansible.builtin.lineinfile:
-        line: SPLUNK_TRACE_URL=https://ingest.fake-realm.signalfx.com/v2/trace
+        line: SPLUNK_TRACE_URL=https://ingest.fake-realm.signalfx.com/v2/trace/otlp
```
Comment (dmitryax) [inline deployments/ansible/molecule/default/verify.yml:48]: This is confusing. Do we remove this env var or replace its value?

File: deployments/ansible/molecule/default/Dockerfile.j2 (hunk 4 above)
Comment (asreehari-splunk) [reply deployments/ansible/molecule/default/Dockerfile.j2:31]: the provo mirror is broken. Jina recommended removing it

**Comment by dmitryax** [comment]
@asreehari-splunk please update the failing tests default_config_test.go:253

File: cmd/otelcol/config/collector/agent_config.yaml [hunk 7]
```diff
@@ -127,9 +127,11 @@ processors:
 
 exporters:
   # Traces
-  sapm:
-    access_token: "${SPLUNK_ACCESS_TOKEN}"
-    endpoint: "${SPLUNK_TRACE_URL}"
+  otlphttp:
+    metrics_endpoint: "${SPLUNK_INGEST_URL}/v2/datapoint/otlp"
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/agent_config.yaml:131]: This is not needed since the exporter isn't used in the metrics pipeline

File: cmd/otelcol/config/collector/ecs_ec2_config.yaml [hunk 8]
```diff
@@ -129,7 +131,7 @@ exporters:
     token: "${SPLUNK_ACCESS_TOKEN}"
     endpoint: "${SPLUNK_INGEST_URL}/v1/log"
     log_data_enabled: false
-
+
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/ecs_ec2_config.yaml:134]:
```suggestion

```

File: tests/zeroconfig/windows/testdata/Dockerfile.iis.server [hunk 9]
```diff
@@ -24,7 +25,7 @@ ENV ACCESS_TOKEN_TMP=$access_token
 RUN `
     $token = $Env:ACCESS_TOKEN_TMP; `
     $collector_msi_path = (dir "c:\setup\splunk-otel-collector-*.msi").FullName; `
-    c:\setup\install.ps1 -access_token $token -msi_path $collector_msi_path -with_dotnet_instrumentation $true -deployment_env zc-iis-test -ingest_url "http://splunk-otel-collector:9943" -trace_url "http://splunk-otel-collector:7276/v2/trace" -network_interface "0.0.0.0"
+    c:\setup\install.ps1 -access_token $token -msi_path $collector_msi_path -with_dotnet_instrumentation $true -deployment_env zc-iis-test -config_path "C:\setup\iis-server-collector.yaml"
```
Comment (dmitryax) [inline tests/zeroconfig/windows/testdata/Dockerfile.iis.server:28]: Is this related to the PR? Can this be moved to a separate PR if not?

**Comment by dmitryax** [comment]
As discussed offline, let's deprecate and remove `SPLUNK_TRACE_URL` env var before proceeding with this change

File: cmd/otelcol/config/collector/ecs_ec2_config.yaml [hunk 10]
```diff
@@ -129,7 +130,6 @@ exporters:
     token: "${SPLUNK_ACCESS_TOKEN}"
     endpoint: "${SPLUNK_INGEST_URL}/v1/log"
     log_data_enabled: false
-
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/ecs_ec2_config.yaml:132]: Please keep the space here

File: cmd/otelcol/config/collector/gateway_config.yaml [hunk 11]
```diff
@@ -1,4 +1,4 @@
-# Configuration file that uses the Splunk exporters (SAPM, SignalFx) to push
+# Configuration file that uses the Splunk exporters (OTLPHTTP, SignalFx) to push
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/gateway_config.yaml:1]:
```suggestion
# Configuration file that uses the Splunk exporters (OTLP HTTP, SignalFx) to push
```

File: cmd/otelcol/config/collector/gateway_config.yaml [hunk 12]
```diff
@@ -91,11 +91,10 @@ processors:
 
 exporters:
   # Traces
-  sapm:
-    access_token: "${SPLUNK_ACCESS_TOKEN}"
-    endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com/v2/trace"
-    sending_queue:
-      num_consumers: 32
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/gateway_config.yaml:98]: This should be kept

File: deployments/nomad/otel-gateway.nomad [hunk 13]
```diff
@@ -194,6 +194,11 @@ exporters:
     ingest_url: https://ingest.${SPLUNK_REALM}.signalfx.com
   debug:
     verbosity: detailed
+  otlphttp:
+    metrics_endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com/v2/datapoint/otlp"
```
Comment (dmitryax) [inline deployments/nomad/otel-gateway.nomad:198]: No ned to define metrics_endpoint

File: tests/zeroconfig/windows/README.md [hunk 14]
```diff
@@ -6,7 +6,7 @@ a deployment following the Zero Configuration procedures.
 
 ## Requirements to run the tests locally
 
-The GitHub job `dotnet-zeroconfig-e2e-test` in the [`build-and-test`](../../../.github/workflows/build-and-test.yml)
+The GitHub job `dotnet-zeroconfig-e2e-test` in the [`win-package-test`](../../../.github/workflows/win-package-test.yml)
```
Comment (dmitryax) [inline tests/zeroconfig/windows/README.md:9]: How is this related?

**Comment by dmitryax** [comment]
Please add a changelog item and a section in https://github.com/signalfx/splunk-otel-collector?tab=readme-ov-file#upgrade-guidelines

File: tests/zeroconfig/windows/README.md (hunk 14 above)
Comment (asreehari-splunk) [reply tests/zeroconfig/windows/README.md:9]: some of the changes from Palo that included this

Comment (dmitryax) [reply tests/zeroconfig/windows/README.md:9]: The fix is valid but it's not related to this PR. I moved it to https://github.com/signalfx/splunk-otel-collector/pull/5703

File: README.md [hunk 15]
```diff
@@ -176,6 +176,21 @@ manually before the backward compatibility is dropped. For every configuration u
 [the default agent config](https://github.com/signalfx/splunk-otel-collector/blob/main/cmd/otelcol/config/collector/agent_config.yaml)
 as a reference.
 
+### From 0.114.0 to 0.115.0
+
+- Replace the sapm exporter in the traces pipline with otlphttp.
```
Comment (dmitryax) [inline README.md:181]:
```suggestion
- Replace the sapm exporter in the traces pipeline with otlphttp.
```

File: README.md [hunk 16]
```diff
@@ -176,6 +176,21 @@ manually before the backward compatibility is dropped. For every configuration u
...
+  
+  More details: https://github.com/signalfx/splunk-otel-collector/pull/5588.
+  
+  Configuration for `otlphttp` exporter should look like this:
```
Comment (dmitryax) [inline README.md:185]: Should be something like this
```suggestion
1. Replace `sapm` exporter with `otlphttp` having the following configuration:
```

File: README.md [hunk 17]
```diff
@@ -176,6 +176,21 @@ manually before the backward compatibility is dropped. For every configuration u
...
+
+  ```yaml
+  otlphttp:
+    traces_endpoint: "${SPLUNK_INGEST_URL}/v2/trace/otlp"
+    headers:
+      "X-SF-Token": "${SPLUNK_ACCESS_TOKEN}"
+  ```
+
```
Comment (dmitryax) [inline README.md:193]: 2. Update `traces` pipeline to use `otlphttp` exporter instead of `sapm`:

```yaml
service:
  pipelines:
    traces:
       exporters: [otlphttp, signalfx]
```

File: README.md [hunk 18]
```diff
@@ -176,6 +176,21 @@ manually before the backward compatibility is dropped. For every configuration u
 [the default agent config](https://github.com/signalfx/splunk-otel-collector/blob/main/cmd/otelcol/config/collector/agent_config.yaml)
 as a reference.
 
+### From 0.114.0 to 0.115.0
+
+- Replace the sapm exporter in the traces pipline with otlphttp.
+  
+  More details: https://github.com/signalfx/splunk-otel-collector/pull/5588.
```
Comment (dmitryax) [inline README.md:183]: Also, mention that `samp` exporter is still working as before, it's just deprecated. We need to make it sound that it's not super critical to update right away.

File: cmd/otelcol/config/collector/ecs_ec2_config.yaml [hunk 19]
```diff
@@ -129,7 +130,7 @@ exporters:
...
+
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/ecs_ec2_config.yaml:133]:
```suggestion

```

File: cmd/otelcol/config/collector/full_config_linux.yaml [hunk 20]
```diff
@@ -637,7 +637,11 @@ exporters:
   # WARN: https://ingest.${SPLUNK_REALM}.signalfx.com does NOT support gRPC at this time; use otlphttp exporter instead
   otlp:
     endpoint: otelcol2:4317
-
+  sapm:
+    access_token: "${SPLUNK_ACCESS_TOKEN}"
+    endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com/v2/trace"
+    sending_queue:
+      num_consumers: 32
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/full_config_linux.yaml:644]: Why is this added?

File: cmd/otelcol/fips/config/agent_config.yaml
Comment (dmitryax) [inline cmd/otelcol/fips/config/agent_config.yaml:1]: Please update other configs in `fips/config` directory

File: deployments/nomad/otel-agent.nomad [hunk 21]
```diff
@@ -212,6 +212,10 @@ exporters:
     sync_host_metadata: true
   debug:
     verbosity: detailed
+  otlphttp:
+    traces_endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com/v2/trace/otlp"
+    headers:
+      "X-SF-Token": "${SPLUNK_ACCESS_TOKEN}"
```
Comment (dmitryax) [inline deployments/nomad/otel-agent.nomad:218]: Please do the other changes

File: deployments/nomad/otel-gateway.nomad [hunk 22]
```diff
@@ -194,6 +194,10 @@ exporters:
     ingest_url: https://ingest.${SPLUNK_REALM}.signalfx.com
   debug:
     verbosity: detailed
+  otlphttp:
+    traces_endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com/v2/trace/otlp"
+    headers:
+      "X-SF-Token": "${SPLUNK_ACCESS_TOKEN}"
```
Comment (dmitryax) [inline deployments/nomad/otel-gateway.nomad:200]: Same here

**Comment by dmitryax** [comment]
Configs in `packaging/technical-addon` also need to be updated

File: deployments/nomad/otel-gateway.nomad (hunk 22 above)
Comment (asreehari-splunk) [reply deployments/nomad/otel-gateway.nomad:200]: similar to agent-config.yaml, keeping sapm in the exporters but not in the pipeline

File: cmd/otelcol/config/collector/gateway_config.yaml [hunk 23]
```diff
@@ -96,6 +96,10 @@ exporters:
     endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com/v2/trace"
     sending_queue:
       num_consumers: 32
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/gateway_config.yaml:98]: Let's remove sapm exporter from here. It's not needed anymore

File: deployments/nomad/otel-gateway.nomad (hunk 22 above)
Comment (dmitryax) [reply deployments/nomad/otel-gateway.nomad:200]: Why do we need to keep it?

**Comment by dmitryax** [comment]
TA configs still need to be updated

File: cmd/otelcol/config/collector/gateway_config.yaml (hunk 23 above)
Comment (asreehari-splunk) [reply cmd/otelcol/config/collector/gateway_config.yaml:98]: ok. Does that apply to the fips/config/gateway_config.yaml

**Comment by dmitryax** [comment]
`cmd/otelcol/config/collector/full_config_linux.yaml` and `packaging/technical-addon/packaging-scripts/cicd-tests/mysql/ta-agent-config.yaml` still not updated

File: deployments/nomad/otel-gateway.nomad (hunk 22 above)
Comment (dmitryax) [reply deployments/nomad/otel-gateway.nomad:200]: Please remove sapm exporter here

Comment (dmitryax) [reply deployments/nomad/otel-gateway.nomad:200]: It's still there

**Review by dmitryax — APPROVED** [review]
